docs/data/data-grid/server-side-data/lazy-loading.md

```diff
@@ -2,14 +2,111 @@
+
+If the user scrolls too fast, the grid loads multiple pages with one request (by adjusting `start` and `end` param) in order to reduce the server load.
+
+The demo below shows the viewport loading mode.
+
+{{"demo": "ServerSideLazyLoadingViewport.js", "bg": "inline"}}
+
+:::info
+The data source demos use a utility function `useMockServer` to simulate the server-side data fetching.
+In a real-world scenario, you should replace this with your own server-side data-fetching logic.
+
+Open info section of the browser console to see the requests being made and the data being fetched in response.
+:::
+
+### Request throttling
+
+While user is scrolling through the grid, rendering context changes and the Data Grid tries to fill in any missing rows by making a new server request.
+The Data Grid throttles new data fetches on the rendering context change to avoid doing unnecessary requests.
+The default throttle time is 500 milliseconds, use `unstable_lazyLoadingRequestThrottleMs` prop to customize it, as the following example demonstrates.
+
+{{"demo": "ServerSideLazyLoadingRequestThrottle.js", "bg": "inline"}}
+
+## Infinite loading
+
+The infinite loading mode is enabled when the row count is unknown (`-1` or `undefined`). New page is loaded when the scroll reaches the bottom of the viewport area.
+
+The area which triggers the new request can be changed using `scrollEndThreshold`.
+
+The demo below shows the infinite loading mode. Page size is set to `15` and the mock server is configured to return a total of `100` rows. Once the response does not contain any new rows, the grid stops requesting new data.
+
+{{"demo": "ServerSideLazyLoadingInfinite.js", "bg": "inline"}}
+
+## Updating the loading mode
+
+The grid changes the loading mode dynamically if the total row count gets updated in any of the three ways described above.
+
+Based on the previous and the new value for the total row count, the following scenarios are possible:
+
+- **Unknown `rowCount` to known `rowCount`**: When the row count is set to a valid value from an unknown value, the Data Grid switches to the viewport loading mode. It checks the number of already fetched rows and adds skeleton rows to match the provided row count.
+
+- **Known `rowCount` to unknown `rowCount`**: If the row count is updated and set to `-1`, the Data Grid resets, fetches the first page, and sets itself in the infinite loading mode.
+
+- **Known `rowCount` greater than the actual row count**: This can happen either by reducing the value of the row count after more rows were already fetched or if the row count was unknown and the grid in the inifite loading mode already fetched more rows. In this case, the grid resets, fetches the first page and continues in one of the modes depending on the new value of the `rowCount`.
```

Comment:
```suggestion
- **Known `rowCount` greater than the actual row count**: This can happen either by reducing the value of the row count after more rows were already fetched, or if the row count was unknown and the Grid (while in the infinite loading mode) already fetched more rows. In this case, the Grid resets, fetches the first page, and then continues in one mode or the other depending on the new value of the `rowCount`.
```